File: Heat/Controllers/PlantClassesController.vb
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Data.Entity
Imports System.Linq
Imports System.Net
Imports System.Web
Imports System.Web.Mvc
Imports Heat
Imports Heat.Repositories

Namespace Controllers
    <Authorize> _
    Public Class PlantClassesController
        Inherits System.Web.Mvc.Controller

        Private db As New HeatDBContext

        ' GET: PlantClasses
        Function Index() As ActionResult
            Return View(db.PlantClasses.ToList())
        End Function

        ' GET: PlantClasses/Details/5
        Function Details(ByVal id As Integer?) As ActionResult
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim plantClass As PlantClass = db.PlantClasses.Find(id)
            If IsNothing(plantClass) Then
                Return HttpNotFound()
            End If
            Return View(plantClass)
        End Function

        ' GET: PlantClasses/Create
        Function Create() As ActionResult
            Return View()
        End Function

        ' POST: PlantClasses/Create
        'To protect from overposting attacks, please enable the specific properties you want to bind to, for 
        'more details see http://go.microsoft.com/fwlink/?LinkId=317598.
        <HttpPost()>
        <ValidateAntiForgeryToken()>
        Function Create(<Bind(Include:="ID,Name")> ByVal plantClass As PlantClass) As ActionResult
            If ModelState.IsValid Then
                db.PlantClasses.Add(plantClass)
                db.SaveChanges()
                Return RedirectToAction("Index")
            End If
            Return View(plantClass)
        End Function

        ' GET: PlantClasses/Edit/5
        Function Edit(ByVal id As Integer?) As ActionResult
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim plantClass As PlantClass = db.PlantClasses.Find(id)
            If IsNothing(plantClass) Then
                Return HttpNotFound()
            End If
            Return View(plantClass)
        End Function

        ' POST: PlantClasses/Edit/5
        'To protect from overposting attacks, please enable the specific properties you want to bind to, for 
        'more details see http://go.microsoft.com/fwlink/?LinkId=317598.
        <HttpPost()>
        <ValidateAntiForgeryToken()>
        Function Edit(<Bind(Include:="ID,Name")> ByVal plantClass As PlantClass) As ActionResult
            If ModelState.IsValid Then
                db.Entry(plantClass).State = EntityState.Modified
                db.SaveChanges()
                Return RedirectToAction("Index")
            End If
            Return View(plantClass)
        End Function

        ' GET: PlantClasses/Delete/5
        Function Delete(ByVal id As Integer?) As ActionResult
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim plantClass As PlantClass = db.PlantClasses.Find(id)
            If IsNothing(plantClass) Then
                Return HttpNotFound()
            End If
            Return View(plantClass)
        End Function

        ' POST: PlantClasses/Delete/5
        <HttpPost()>
        <ActionName("Delete")>
        <ValidateAntiForgeryToken()>
        Function DeleteConfirmed(ByVal id As Integer) As ActionResult
            Dim plantClass As PlantClass = db.PlantClasses.Find(id)
            db.PlantClasses.Remove(plantClass)
            db.SaveChanges()
            Return RedirectToAction("Index")
        End Function

        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            If (disposing) Then
                db.Dispose()
            End If
            MyBase.Dispose(disposing)
        End Sub
    End Class
End Namespace

File: Heat.ConvertedToC#/Controllers/PlantClassesController.cs
using System.Linq;
using System.Web.Mvc;
using System.Data.Entity;
using System.Net;
using Heat.Repositories;
using Heat.Models;

namespace Heat.Controllers
{
    [Authorize()]
	public class PlantClassesController : System.Web.Mvc.Controller
	{


		private IHeatDBContext _db;

        public PlantClassesController(IHeatDBContext context)
        {
            _db = context;
        }

		// GET: PlantClasses
		public ActionResult Index()
		{
			return View(_db.PlantClasses.ToList());
		}

		// GET: PlantClasses/Details/5
		public ActionResult Details(int? id)
		{
			if ((id == null)) {
				return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
			}
			PlantClass plantClass = _db.PlantClasses.Find(id);
			if ((plantClass == null)) {
				return HttpNotFound();
			}
			return View(plantClass);
		}

		// GET: PlantClasses/Create
		public ActionResult Create()
		{
			return View();
		}

		// POST: PlantClasses/Create
		//To protect from overposting attacks, please enable the specific properties you want to bind to, for 
		//more details see http://go.microsoft.com/fwlink/?LinkId=317598.
		[HttpPost()]
		[ValidateAntiForgeryToken()]
		public ActionResult Create(		[Bind(Include = "ID,Name")]
PlantClass plantClass)
		{
			if (ModelState.IsValid) {
				_db.PlantClasses.Add(plantClass);
				_db.SaveChanges();
				return RedirectToAction("Index");
			}
			return View(plantClass);
		}

		// GET: PlantClasses/Edit/5
		public ActionResult Edit(int? id)
		{
			if ((id == null)) {
				return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
			}
			PlantClass plantClass = _db.PlantClasses.Find(id);
			if ((plantClass == null)) {
				return HttpNotFound();
			}
			return View(plantClass);
		}

		// POST: PlantClasses/Edit/5
		//To protect from overposting attacks, please enable the specific properties you want to bind to, for 
		//more details see http://go.microsoft.com/fwlink/?LinkId=317598.
		[HttpPost()]
		[ValidateAntiForgeryToken()]
		public ActionResult Edit(		[Bind(Include = "ID,Name")]
PlantClass plantClass)
		{
			if (ModelState.IsValid) {
				_db.SetModified(plantClass) ;
				_db.SaveChanges();
				return RedirectToAction("Index");
			}
			return View(plantClass);
		}

		// GET: PlantClasses/Delete/5
		public ActionResult Delete(int? id)
		{
			if ((id == null)) {
				return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
			}
			PlantClass plantClass = _db.PlantClasses.Find(id);
			if ((plantClass == null)) {
				return HttpNotFound();
			}
			return View(plantClass);
		}

		// POST: PlantClasses/Delete/5
		[HttpPost()]
		[ActionName("Delete")]
		[ValidateAntiForgeryToken()]
		public ActionResult DeleteConfirmed(int id)
		{
			PlantClass plantClass = _db.PlantClasses.Find(id);
			_db.PlantClasses.Remove(plantClass);
			_db.SaveChanges();
			return RedirectToAction("Index");
		}

		protected override void Dispose(bool disposing)
		{
			if ((disposing)) {
				_db.Dispose();
			}
			base.Dispose(disposing);
		}
	}
}
